### PR TITLE
Add template files with ha-sap-terraform-deployments project

### LINKED ISSUE #15: Add NFS HA cluster configuration template and split brain script
By now, we don't ship the HA NFS share configuration template in this package (as we do in other formulas).
I think that using HA DRBD to serve a NFS share is really common, so having the cluster template here makes total sense.
We have received some feedback from customers, that they don't have the cluster configuration, and I think having it here makes sense, than copy/pasting from other sources.

By now, these files are stored in:

[NFS cluster resources](https://github.com/SUSE/ha-sap-terraform-deployments/blob/develop/salt/drbd_node/files/templates/drbd_cluster.j2)  (I would call it `nfs_cluster_resources.j2`)
[Split brain script](https://github.com/SUSE/ha-sap-terraform-deployments/blob/develop/salt/drbd_node/files/notify-split-brain-haclusterexporter-suse-metric.sh)

@nick-wang What do you think?

### PR DESCRIPTION
**Pull request by nick-wang**
1. Polish the existed template and spec file.

2. Add the DRBD/NFS templates with ha-sap-terraform-deployments.
    Support libvirt/gcp/azure/aws provider.
    Leave the [cluster examples](https://github.com/SUSE/drbd-formula/tree/master/examples/with_pacemaker) since DRBD HA cluster may run without ha-sap-terraform-deployments
    
    Fix https://github.com/SUSE/drbd-formula/issues/15

### CODE REVIEW

**Comment by nick-wang** [comment]
The new pillar files/template are come from https://github.com/SUSE/ha-sap-terraform-deployments/releases/tag/7.0.0
Plan to remove the template(leave the pillar files in automatic) from ha-sap-terraform-deployments after merged.

File: drbd-formula.spec
Comment (arbulu89) [inline drbd-formula.spec]: The reason why we added this `if` statement is that before SLE12SP3 the `license` tag doesn't exist.
Actually, as we don't build anymore for versions less than SLE12SP3, I would remove this. We would have:
```
%doc README.md
%license LICENSE

```

File: examples/with_ha-sap-terraform-deployments/pillar.example.cluster_cloud
Comment (arbulu89) [inline examples/with_ha-sap-terraform-deployments/pillar.example.cluster_cloud]: I wouldn't personally add this file.
It has a lot of references to the `ha-sap-terraform-deployments` project. Actually, it would only work in that project. I would just keep this in that project

**Review by arbulu89 — CHANGES_REQUESTED** [review]
@nick-wang The idea is good. I have some things that I would prefer to change.

- I wouldn't add the `pillar.example.cluster_cloud` file. It doesn't belong to this project in my opinion
- I would rename some templates:
  1. `examples/with_ha-sap-terraform-deployments/drbd_with_nfs_cloud.j2` -> `examples/cluster_resources_nfs_cloud.j2`
  2. `examples/with_ha-sap-terraform-deployments/pillar.example.drbd_cloud` -> `pillar.example`
  3. `examples/with_pacemaker/drbd_with_nfs_crm.js` -> examples/cluster_resources_nfs.j2`

I guess that the cloud and non-cloud templates are created for simplicity, right? So if we use libvirt for example, we would need to use the non cloud version

File: drbd-formula.spec
Comment (nick-wang) [reply drbd-formula.spec]: I made a mistake. Follow the spec cleaner without check too much, my bad to make the regression.

File: drbd-formula.spec
Comment (arbulu89) [inline drbd-formula.spec]: As commented, I would take advantage of these fixes and remove the `< 120300` condition. We don't build anymore for versions less than SLE12SP3, and maintenance has been a bit picky in these things.
The next would be just fine:
```
%doc README.md
%license LICENSE
```
But well, we can do this any other time

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hi @nick-wang ,

I still don't like the folder naming. I wouldn't put references to `ha-sap-terraform-deployments`. This makes the user think that this templates are only for that project, and that's not true.
I would just simply remove the `with_ha-sap-terraform-deployments` and `with_pacemaker` folders. I would put all the templates directly in the `templates` folder.
In fact, I would rename `examples` by `templates`

File: drbd-formula.spec
Comment (nick-wang) [reply drbd-formula.spec]: Got it, i will fix it this time.

**Comment by nick-wang** [comment]
> I still don't like the folder naming. I wouldn't put references to `ha-sap-terraform-deployments`. This makes the user think that this templates are only for that project, and that's not true.
> I would just simply remove the `with_ha-sap-terraform-deployments` and `with_pacemaker` folders. I would put all the templates directly in the `templates` folder.
> In fact, I would rename `examples` by `templates`

@arbulu89 The reason why i split `with_ha-sap-terraform-deployments` folder is because the template `cluster_resources_nfs_cloud.j2` and corresponding pillar file has too many elements of `cloud provider`, which may confused user and add complexity when customized the configuration. Though it is actually have condition to handler the case that `cloud_provider` not in `[aws, gcp, azure]`.

I rename the `with_ha-sap-terraform-deployments` to `with_pacemaker_cloud`. If you still think it is not good to keep the template(https://github.com/nick-wang/drbd-formula/blob/issue-15/examples/with_pacemaker/cluster_resources_nfs.j2). I can merge the folders of `with pacemaker`. Have one with `nfs formula` and the other one with `pacemaker habootstrap formula`

Will rename `examples` to `templates`

**Review by arbulu89 — APPROVED** [review]
@nick-wang 
I don't still like the naming, but we can go with it no issues.
If you don't want to put them in the same folder, let's at least rename too:
- `with_ha-sap-terraform-deployments/cluster_resources_nfs_cloud.j2` -> `habootstrap-formula/cluster_resources_nfs_cloud.j2`
- `with_ha-sap-terraform-deployments/notify-split-brain-haclusterexporter-suse-metric.sh` -> `ha_cluster_exporter/notify-split-brain-haclusterexporter-suse-metric.sh`
- (Both in `templates` folder). I don't want the `ha-sap-terraform-deployments` reference, because this is just a casuality, the templates are to be used by the `habootstrap-formula` and the `ha_cluster_exporter`.
**PD: I see more and more a lot of confusion having the cluster resources in the formulas that are not the cluster formula. I'm considering to move all the cluster resource templates there. In the past, they were put in these folders for some business reason, but I think they don't make sense. We will need to discuss this**

**Comment by nick-wang** [comment]
> @nick-wang
> I don't still like the naming, but we can go with it no issues.
> If you don't want to put them in the same folder, let's at least rename too:
> 
> * `with_ha-sap-terraform-deployments/cluster_resources_nfs_cloud.j2` -> `habootstrap-formula/cluster_resources_nfs_cloud.j2`
> * `with_ha-sap-terraform-deployments/notify-split-brain-haclusterexporter-suse-metric.sh` -> `ha_cluster_exporter/notify-split-brain-haclusterexporter-suse-metric.sh`
> * (Both in `templates` folder). I don't want the `ha-sap-terraform-deployments` reference, because this is just a casuality, the templates are to be used by the `habootstrap-formula` and the `ha_cluster_exporter`.
>   **PD: I see more and more a lot of confusion having the cluster resources in the formulas that are not the cluster formula. I'm considering to move all the cluster resource templates there. In the past, they were put in these folders for some business reason, but I think they don't make sense. We will need to discuss this**

@arbulu89 Thanks a lot for the idea.
I do agree with `habootstrap-formula` sounds better.

I changed the folder structure again, now is like:
```
templates # tree
.
├── habootstrap-formula
│   ├── cluster_resources_nfs_cloud.j2
│   ├── cluster_resources_nfs.j2
│   └── pillar.example.cluster
├── ha_cluster_exporter
│   └── notify-split-brain-haclusterexporter-suse-metric.sh
├── nfs-formula
│   └── pillar.example.nfs
├── pillar.example.drbd
└── README.md
```
How do you think of it?

**Review by arbulu89 — APPROVED** [review]
Great! Thanks @nick-wang

**Comment by nick-wang** [comment]
@arbulu89 Thanks for the help. I will do some test(build/install) before click merge. (However, i think it should be fine)

**Comment by arbulu89** [comment]
> @arbulu89 Thanks for the help. I will do some test(build/install) before click merge. (However, i think it should be fine)

Nice! Keep in mind that afterwards you will need to update the `ha-sap-terraform-deployments` project to use this new structure and remove duplicated files XD

**Comment by nick-wang** [comment]
> > @arbulu89 Thanks for the help. I will do some test(build/install) before click merge. (However, i think it should be fine)
> 
> Nice! Keep in mind that afterwards you will need to update the `ha-sap-terraform-deployments` project to use this new structure and remove duplicated files XD

@arbulu89 Yes, i was plan to test this with changed `ha-sap-terraform-deployments`. Only merge this after all finished.

Will adapt(remove) the j2 template and exporter script with the one in `drbd-formula`. Only leave the pillar file in `automatic` for pre_deployment. The two change should be used together with v8. I think besides the release note of `ha-sap-terraform-deployments`, also better to maintain a `requirements.txt` to list the dependencies. 
How do you think?

**Comment by nick-wang** [comment]
Corresponding change in `ha-sap-terraform-deployments` project is https://github.com/SUSE/ha-sap-terraform-deployments/pull/663

**Comment by nick-wang** [comment]
From this pull request, any change of the path of template need to align with project `ha-sap-terraform-deployments`.
Refer to https://github.com/SUSE/ha-sap-terraform-deployments/pull/663

**Pull request merged**
